plugin-scalebar/src/main/java/com/mapbox/maps/plugin/scalebar/ScaleBarImpl.kt

```diff
@@ -294,11 +301,33 @@ class ScaleBarImpl : ScaleBar, View {
     }
   }
 
-  private fun drawText(canvas: Canvas, text: String, x: Float, y: Float) {
+  private fun drawText(canvas: Canvas, text: String, x: Float, y: Float, lastUnitBarText: Boolean) {
+    var safeX = x
+
+    // As an optimization only check if it goes beyond right view for the last unit bar text
+    if (lastUnitBarText) {
+      // Check if it goes beyond the right margin of the view
+      val textWidthPx = textPaint.measureText(text)
+      val textMaxRightPx = x + strokePaint.strokeWidth + when (textPaint.textAlign) {
```

Comment:
As far as I understood `strokeWidth` was the main reason for the text being cut?

Reply:
No, the main reason was that the `x` we are passing is based on the position of the current unit bar and that sometimes makes it grow beyond the right margin

Reply:
But `strokeWidth` needs to be taken into account also, since `measureText` doesn't.